naxsi_src/naxsi_utils.c

```diff
@@ -665,132 +774,148 @@ ngx_http_dummy_create_hashtables_n(ngx_http_dummy_loc_conf_t *dlc,
-	else
-	  ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "whitelists has no target uri.");
-	return (NGX_ERROR);
+        ngx_conf_log_error(
+            NGX_LOG_EMERG, cf, 0,
+            "Following whitelist doesn't target any zone or is incorrect :");
+        if (name_idx != -1)
+          ngx_conf_log_error(
+              NGX_LOG_EMERG, cf, 0, "whitelist target name : %V",
+              &(custloc_array(curr_r->br->custom_locations->elts)[name_idx]
+                    .target));
+        else
+          ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                             "whitelist has no target name.");
+        if (uri_idx != -1)
+          ngx_conf_log_error(
+              NGX_LOG_EMERG, cf, 0, "whitelist target uri : %V",
+              &(custloc_array(curr_r->br->custom_locations->elts)[uri_idx]
+                    .target));
+        else
+          ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                             "whitelists has no target uri.");
+        return (NGX_ERROR);
       } /* LCOV_EXCL_STOP */
-    curr_r->br->zone = zone;
-    /*
-    ** Handle regular-expression-matchzone rules :
-    ** Store them in a separate linked list, parsed
-    ** at runtime.
-    */
-    if (curr_r->br->rx_mz == 1) {
-      if (!dlc->rxmz_wlr) {
-	dlc->rxmz_wlr = ngx_array_create(cf->pool, 1,
-					 sizeof(ngx_http_rule_t *));
-	if (!dlc->rxmz_wlr) return (NGX_ERROR); /* LCOV_EXCL_LINE */
-      }
-      if (name_idx != -1 && !custloc_array(curr_r->br->custom_locations->elts)[name_idx].target_rx) {
-	custloc_array(curr_r->br->custom_locations->elts)[name_idx].target_rx = 
-	  ngx_pcalloc(cf->pool, sizeof(ngx_regex_compile_t));
-	rgc = custloc_array(curr_r->br->custom_locations->elts)[name_idx].target_rx;
-	rgc->options = PCRE_CASELESS|PCRE_MULTILINE;
-	rgc->pattern = custloc_array(curr_r->br->custom_locations->elts)[name_idx].target;
-	rgc->pool = cf->pool;
-	rgc->err.len = 0;
-	rgc->err.data = NULL;
-	//custloc_array(curr_r->br->custom_locations->elts)[name_idx].target;
-	if (ngx_regex_compile(rgc) != NGX_OK)
-	  return (NGX_ERROR);
+      curr_r->br->zone = zone;
+      /*
+      ** Handle regular-expression-matchzone rules :
+      ** Store them in a separate linked list, parsed
+      ** at runtime.
+      */
+      if (curr_r->br->rx_mz == 1) {
+        if (!dlc->rxmz_wlr) {
+          dlc->rxmz_wlr =
+              ngx_array_create(cf->pool, 1, sizeof(ngx_http_rule_t *));
+          if (!dlc->rxmz_wlr)
+            return (NGX_ERROR); /* LCOV_EXCL_LINE */
+        }
+        if (name_idx != -1 &&
+            !custloc_array(curr_r->br->custom_locations->elts)[name_idx]
+                 .target_rx) {
+          custloc_array(curr_r->br->custom_locations->elts)[name_idx]
+              .target_rx = ngx_pcalloc(cf->pool, sizeof(ngx_regex_compile_t));
+          rgc = custloc_array(curr_r->br->custom_locations->elts)[name_idx]
+                    .target_rx;
```

Comment:
maybe check the value of the pointer  `rgc` here?

Reply:
Done